### PR TITLE
Fix leaks in make_input_stream()

### PR DESCRIPTION
**Pull request by tobim**

### CODE REVIEW

**Comment by tobim** [comment]
I personally prefer a naked `new` vs `unique_ptr::release()` because the latter looks so innocent where as any experienced c++ programmer knows to be careful around the `new`.

Otoh moving the `release()` into the constructor makes it so that the pointers ownership is reflected in the type system at all times. I'll apply the change.

File: libvast/src/detail/make_io_stream.cpp [hunk 1]
```diff
@@ -25,6 +25,14 @@ namespace detail {
 
 expected<std::unique_ptr<std::istream>>
 make_input_stream(const std::string& input, bool is_uds) {
+  struct owning_istream : public std::istream {
+    owning_istream(std::unique_ptr<std::streambuf>&& ptr) : std::istream{ptr.release()} {
```
Comment (mavam) [inline libvast/src/detail/make_io_stream.cpp]: Style: this line looks like it's exceeding 80 characters.

File: libvast/src/detail/make_io_stream.cpp [hunk 2]
```diff
@@ -35,15 +43,15 @@ make_input_stream(const std::string& input, bool is_uds) {
                         "failed to connect to UNIX domain socket at", input);
     auto remote_fd = uds.recv_fd(); // Blocks!
     auto sb = std::make_unique<fdinbuf>(remote_fd);
-    return std::make_unique<std::istream>(sb.release());
+    return std::make_unique<owning_istream>(std::move(sb));
   }
   if (input == "-") {
-    auto sb = std::make_unique<fdinbuf>(0); // stdin
-    return std::make_unique<std::istream>(sb.release());
+    auto sb = std::make_unique<fdinbuf>(0);
```
Comment (mavam) [inline libvast/src/detail/make_io_stream.cpp]: Why did you remove the comment? I think it's useful reminder for the reader that FD 0 is standard input.
